### ISSUE TITLE
Authentication not work with Windows Phone 8.1 (universal)

### ISSUE THREAD

**Issue by nhat-tong** [adrianchia/auth0-ionic-adfs-sample#1]
Hi Adrianchia,

I'am a newbie on development the applications with Cordova. So, I'm really happy when i found your work with the problem of auth-dialog and in app browser for authenticate with ADFS. Today, I deploy my application on Windows Phone 8.1 (universal) simulator for tester. Unfortunately, the authentication did'nt work. I received 401 error unauthoried comme beginning. 

So, can you tell me how to fix this prob in Windows Phone Universal, please ?

Thanks so much,

Nhat

**Comment by adrianchia** [comment]
Hi Nhat,

I don't have Windows 8.1 development environment so I can't tell you for sure, but basically it drills down to whether 
1. [cordova-wp8](https://github.com/apache/cordova-windows) supports auth dialog supports auth dialog for Windows Universal app
2. [cordova-inapp-browser](https://github.com/apache/cordova-plugin-inappbrowser) plugin supports auth dialog for Windows Universal app

Edit: 1 should be cordova-windows instead of wp8

**Comment by nhat-tong** [comment]
Hi Adrianchia,

Thanks for your anwser. But when I installed cordova-inapp-browser, the pop-up ADFS for username/pass didn't display any more on Android. I had also 401 error unauthorized (it worked well with your custom plugin). I think that InAppBrowser of cordova didn't support yet auth-dialog plugin.

More information, your custom plugin resolved my problem of redirection with ADFS on Android (I just installed two plugins as you proprosed). When the redirection ADFS returned, it called auth-dialog plugin for display the pop-up username/pass.
Unfortunately, this behavior did'nt work with Windows Phone 8 (I tested on simulator and on device) and Windows Phone 8.1. I hadn't an iPhone for verify this behavior.

I try to read your InAppBrowser custom plugin code but I didn't found the code which call auth-dialog when the redirection ADFS returned. I wanna to extend this plugin for support Windows Phone 8 and Windows 8.1.

Call you help me, please ?

Thank you,

Nhat

**Comment by adrianchia** [comment]
Yes, the current InAppBrowser plugin does not yet. This is mentioned in Cordova's Issue Tracking System https://issues.apache.org/jira/browse/CB-8534 and also https://github.com/apache/cordova-plugin-inappbrowser/pull/82. The reason why it worked in Android with Custom Code is that Cordova-Android has added support for auth dialog, which is based on Android WebView (see [here](https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/engine/SystemWebViewClient.java#L87)). This allows plugins to implement auth dialogs. I am not sure about how Windows Universal App does this but it could break down into several scenarios:
1. Can a native Windows Universal App access a page secured by ADFS with an auth dialog (without cordova)? Note that you may have to implement the auth dialog yourself, the other way to put this is does it support any way to add a listener / handler so that you can implement an auth dialog?
2. If 1 is possible, can cordova-windows support the same mechanism?
3. If 2 is also possible it is most likely that cordova-inapp-browser needs to add support for that.
